src/authorization/UnionPermissionReader.ts

```diff
@@ -12,20 +13,34 @@ export class UnionPermissionReader extends UnionHandler<PermissionReader> {
     super(readers);
   }
 
-  protected async combine(results: PermissionSet[]): Promise<PermissionSet> {
-    const result: PermissionSet = {};
-    for (const permissionSet of results) {
-      for (const [ key, value ] of Object.entries(permissionSet) as [ CredentialGroup, Permission | undefined ][]) {
-        result[key] = this.applyPermissions(value, result[key]);
-      }
+  protected async combine(results: PermissionMap[]): Promise<PermissionMap> {
+    const result: PermissionMap = new IdentifierMap();
+    for (const permissionMap of results) {
+      this.mergePermissionMaps(permissionMap, result);
     }
     return result;
   }
 
+  /**
+   * Merges all entries of the given map into the result map.
+   */
+  private mergePermissionMaps(permissionMap: PermissionMap, result: PermissionMap): void {
+    for (const [ identifier, permissionSet ] of permissionMap) {
+      for (const [ credential, permission ] of Object.entries(permissionSet) as [CredentialGroup, Permission][]) {
+        let resultSet = result.get(identifier);
+        if (!resultSet) {
+          resultSet = {};
+          result.set(identifier, resultSet);
+        }
```

Comment:
```suggestion
        let resultSet = result.get(identifier, {});
```
you know you want to 😉

Reply:
I'm more a fun of introducing new utility functions vs diverging from the Map interface, but `getDefault` would be a nice new function in `MapUtil` as that is not an uncommon case.